--- a/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c
+++ b/hal/dp_ops/edma_dp/edma_v1/edma_data_plane.c
@@ -599,10 +599,10 @@ static void edma_if_set_features(struct
 {
 	struct net_device *netdev = dpc->dev;
 
-	netdev->features |= NETIF_F_GRO;
-	netdev->hw_features |= NETIF_F_GRO;
-	netdev->vlan_features |= NETIF_F_GRO;
-	netdev->wanted_features |= NETIF_F_GRO;
+	netdev->features |= NETIF_F_GRO | NETIF_F_HW_CSUM | NETIF_F_RXCSUM;
+	netdev->hw_features |= NETIF_F_GRO | NETIF_F_HW_CSUM | NETIF_F_RXCSUM;
+	netdev->vlan_features |= NETIF_F_GRO | NETIF_F_HW_CSUM | NETIF_F_RXCSUM;
+	netdev->wanted_features |= NETIF_F_GRO | NETIF_F_HW_CSUM | NETIF_F_RXCSUM;
 }
 
 /* TODO - check if this is needed */
